ios/authentication.md

```diff
@@ -785,3 +819,402 @@ Note : `AWSGoogleSignIn` is only needed for using Google Login in your app and `
+
+#### Setup Amazon Cognito Hosted UI in iOS App
+
+1. To configure your application for hosted UI, you need to use *HostedUI* options. Update your `awsconfiguration.json` file to add a new configuration for `MobileClient`. The configuration should look like this:
+
+    ```json
+    {
+        "IdentityManager": {
+            ...
+        },
+        "CredentialsProvider": {
+            ...
+        },
+        "CognitoUserPool": {
+            ...
+        },
+        "MobileClient": {
+            "Default": {
+                "HostedUI": {
+                    "WebDomain": "https://YOUR_AUTH_DOMAIN.auth.us-west-2.amazoncognito.com",
+                    "AppClientId": "YOUR_APP_CLIENT_ID",
+                    "AppClientSecret": "YOUR_APP_CLIENT_SECRET",
+                    "SignInRedirectURI": "myapp://",
+                    "SignOutRedirectURI": "myapp://",
+                    "Scopes": ["openid", "email"]
+                }
+            }
+        }
+    }
+    ```
+
+1. Add `myapp://` to your app's URL schemes:
+
+    Right-click Info.plist and then choose Open As > Source Code.
+
+    Add the following entry in URL scheme:
+
+    ```xml
+        <plist version="1.0">
+
+        <dict>
+        <!-- YOUR OTHER PLIST ENTRIES HERE -->
+
+        <!-- ADD AN ENTRY TO CFBundleURLTypes for Cognito Auth -->
+        <!-- IF YOU DO NOT HAVE CFBundleURLTypes, YOU CAN COPY THE WHOLE BLOCK BELOW -->
+        <key>CFBundleURLTypes</key>
+        <array>
+            <dict>
+                <key>CFBundleURLSchemes</key>
+                <array>
+                    <string>myapp</string>
+                </array>
+            </dict>
+        </array>
+
+        <!-- ... -->
+        </dict>
+    ```
+
+#### Launching the Hosted UI
+
+To launch the Hosted UI from from your application, you can use the `showSignIn` API of `AWSMobileClient.sharedInstance()`:
+
+```swift
+// Optionally override the scopes based on the usecase.
+let hostedUIOptions = HostedUIOptions(scopes: ["openid", "email"])
+
+// Present the Hosted UI sign in.
+AWSMobileClient.sharedInstance().showSignIn(navigationController: self.navigationController!, hostedUIOptions: hostedUIOptions) { (userState, error) in
+    if let error = error as? AWSMobileClientError {
+        print(error.localizedDescription)
+    }
+    if let userState = userState {
+        print("Status: \(userState.rawValue)")
+    }
+}
+```
+
+> Optional: If your app deployment target is < iOS 11, add the following callback in your App Delegate's `application:open url` method:
+
+```swift
+func application(_ application: UIApplication, open url: URL, sourceApplication: String?, annotation: Any) -> Bool {
+    return AWSMobileClient.sharedInstance().interceptApplication(application, open: url, sourceApplication: sourceApplication, annotation: annotation)
+}
+```
+
+Note: By default, the Hosted UI will show all login options; the username-password flow as well as any social providers which are configured.
```

Comment:
Please add something like: If you wish to bypass the extra sign-in screen showing all the provider options, and redirect directly to the social provider login, you can use `HostedUIOptions` as outlined below and attach this to a UI element such as a button in your application.

Reply:
Will update.

Reply:
Done

Reply:
Note: By default, the Hosted UI will show all login options; the username-password flow as well as any social providers which are configured. If you wish to bypass the extra sign-in screen showing all the provider options and launch your desired social provider login directly, you can set the `HostedUIOptions` as shown in the next section.